test/e2e/localvolume_test.go

```diff
@@ -151,6 +159,269 @@ func LocalVolumeTest(ctx *framework.Context, cleanupFuncs *[]cleanupFn) func(*te
+			Name:      name,
+			Namespace: f.Namespace,
+			Labels: map[string]string{
+				"app":     pvConsumerLabel,
+				"pv-name": pv.Name,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app":     pvConsumerLabel,
+						"pv-name": pv.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:  "busybox",
+							Image: "busybox",
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									MountPath: "/data",
+									Name:      "volume-to-debug",
+								},
+							},
+							Command: []string{"/bin/sh", "-c"},
+							Args: []string{
+								"dd if=/dev/random of=/tmp/random.img bs=512 count=1",     // create a new file named random.img
+								"md5VAR1=$(md5sum /tmp/random.img | awk '{ print $1 }')",  // calculate md5sum of random.img
+								"cp /tmp/random.img /data/random.img",                     // copy random.img file to pvc mountpoint
+								"md5VAR2=$(md5sum /data/random.img | awk '{ print $1 }')", // recalculate md5sum of file random.img stored in pvc
+								"if [[ \"$md5VAR1\" != \"$md5VAR2\" ]];then exit 1; fi",   // verifies that the md5sum hasn't changed
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "volume-to-debug",
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: pvc.Name,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// create pvc
+	matcher.Eventually(func() error {
+		t.Logf("creating pvc: %q", pvc.Name)
+		return f.Client.Create(goctx.TODO(), pvc, &framework.CleanupOptions{TestContext: ctx})
+	}, time.Minute, time.Second*2).ShouldNot(gomega.HaveOccurred(), "creating pvc")
+
+	// recording a time before the job was created
+	toRound := time.Now()
+	// rounding down to the same granularity as the timestamp
+	timeStarted := time.Date(toRound.Year(), toRound.Month(), toRound.Day(), toRound.Hour(), toRound.Minute(), 0, 0, toRound.Location())
+
+	// create consuming job
+	matcher.Eventually(func() error {
+		t.Logf("creating job: %q", job.Name)
+		return f.Client.Create(goctx.TODO(), job, &framework.CleanupOptions{TestContext: ctx})
+	}, time.Minute, time.Second*2).ShouldNot(gomega.HaveOccurred(), "creating job")
+
+	// wait for job to complete
+	matcher.Eventually(func() int32 {
+		t.Log("waiting for job to complete")
+		err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: job.Name, Namespace: job.Namespace}, job)
+		if err != nil {
+			t.Logf("error fetching job: %+v", err)
+			return 0
+		}
+		t.Logf("job completions: %d", job.Status.Succeeded)
+		return job.Status.Succeeded
+	}, time.Minute*5, time.Second*2).Should(gomega.BeNumerically(">=", 1), "waiting for job to complete")
+
+	// return pods because they have to be deleted before pv is released
```

Comment:
Do you need to track the Pods? A job with `ttlSecondsAfterFinished: 0` will automatically delete the Pods and checking `job.Status.Succeeded` could be enough.

Reply:
`ttlSecondsAfterFinished` only documents deleting the job. When I checked, deleting the job did not immediately delete the completed job, and we want the job to stick around.

```
	// ttlSecondsAfterFinished limits the lifetime of a Job that has finished
	// execution (either Complete or Failed). If this field is set,
	// ttlSecondsAfterFinished after the Job finishes, it is eligible to be
	// automatically deleted. When the Job is being deleted, its lifecycle
	// guarantees (e.g. finalizers) will be honored. If this field is unset,
	// the Job won't be automatically deleted. If this field is set to zero,
	// the Job becomes eligible to be deleted immediately after it finishes.
	// This field is alpha-level and is only honored by servers that enable the
	// TTLAfterFinished feature.
	// +optional
	TTLSecondsAfterFinished *int32 
```

Reply:
ack